bundles/org.openhab.binding.pjlinkdevice/src/main/java/org/openhab/binding/pjlinkdevice/internal/device/command/CachedCommand.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.pjlinkdevice.internal.device.command;
+
+import java.io.IOException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.cache.ExpiringCache;
+
+/**
+ * Basic command interface allowing to execute the command.
+ *
+ * @author Nils Schnabel - Initial contribution
+ */
+@NonNullByDefault
+public class CachedCommand<ResponseType extends Response<?>> implements Command<ResponseType> {
+
+  private Command<ResponseType> cachedCommand;
+  private ExpiringCache<ResponseType> cache;
+
+  public CachedCommand(Command<ResponseType> cachedCommand) {
+    this(cachedCommand, 1000);
+  }
+
+  public CachedCommand(Command<ResponseType> cachedCommand, int expiry) {
+    this.cachedCommand = cachedCommand;
+    this.cache = new ExpiringCache<ResponseType>(expiry, () -> {
+      try {
+        return this.cachedCommand.execute();
+      } catch (ResponseException | IOException | AuthenticationException e) {
+        throw new RuntimeException(e);
```

Comment:
I'm not sure if there's a better way to do this. Essentially, I'm wrapping these exception types into a RuntimeException because the ExpiringCache does not allow declared exceptions. In[ execute(), I'm catching and unwrapping](https://github.com/openhab/openhab2-addons/pull/6063/files#diff-0c1f98b8a011ddcd130cd2bc1a21e5d5R60) them again.

Reply:
This also (correctly, of course) raises a static code check issue of type AvoidThrowingRawExceptionTypes. So I hope there's another way to solve it...

Reply:
I do not fully understand what you are trying to do and what you need this class for. What shall happen if "fecth new value" fails? I guess something like setting the channel to UNDEF. Wouldn't it be possible to use the `ExpiringCache` directly in the handler and modifiy the update-code so that it return UNDEF?

Reply:
This class is intended to cache the results of commands in a generic way.
Currently it is only used for the LampStatesCommand results, as there might be multiple (configured) channels now all relying on the same command result:

Imagine I have four channels regarding lamp state: 

- lamp1Active/lamp1Hours relating to lamp no. 1 (preconfigured), 
- lamp2Active/lamp2Hours relating to lamp no. 2 (configured manually). 

Now the way PJLinks works is that I send one request and I get information about ALL lamps as a response. 

Now if there is a chain of REFRESH on all of those channels (e.g. during thing initialization), I want to avoid that I immediately request again the same information I received only milliseconds ago. That's why I want to cache the result for one second.

The idea is that this CachedCommand work transparently. It should raise exactly the same exceptions the wrapped command would have raised if there is a problem.

How these exceptions are handled in the end does not change with this PR (can be seen here: https://github.com/openhab/openhab2-addons/blob/master/bundles/org.openhab.binding.pjlinkdevice/src/main/java/org/openhab/binding/pjlinkdevice/internal/PJLinkDeviceHandler.java#L188).